File: py/algorithm/bubble_sort.py
from typing import List


def sort(items: List[int]) -> None:
    for i in range(len(items)):
        swapped = False

        for j in range(len(items) - i - 1):

            if items[j] > items[j + 1]:
                items[j], items[j + 1] = items[j + 1], items[j]
                swapped = True

        if not swapped:
            break


items = [2, 5, 0, 9, -1, 8, 10]
sort(items)
print(items)

File: rust/algorithm/src/bubble_sort.rs
pub fn sort(items: &mut [i8; 10]) -> () {
    let mut swapped = true;

    while swapped {
        swapped = false;

        for i in 0..items.len() - 1 {
            if items[i] > items[i + 1] {
                items.swap(i, i + 1);
                swapped = true;
            }
        }
    }
}
